### PR TITLE
[ANCHOR-206] chore: add docker-compose files to launch services using containers.

### PR DESCRIPTION
**Pull request by paulormnas**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [x] This PR adds tests for the most critical parts of the new functionality or fixes.
Add docker-compose files to run e2e tests on containers

### What

Add docker-compose files to run e2e test with containers.

### Why

The docker-compose files make it easier to run tests during CI.

### Known limitations

All the common-services need to be up and running so the e2e python script can run all the tests.

### CODE REVIEW

File: docker-compose/common-services/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#FROM --platform=linux/amd64 gradle:7.6.0-jdk11-alpine AS build
```
Comment (lijamie98) [inline docker-compose/common-services/Dockerfile:1]: What is the purpose of the Dockerfile? 

Is this the same as [this Dockerfile](https://github.com/stellar/java-stellar-anchor-sdk/blob/legacy/main-1.2/Dockerfile)?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,7 @@
 # Changelog
+## 1.2.12
+* Add docker-compose files to run e2e tests with containers
```
Comment (lijamie98) [inline CHANGELOG.md:3]: We do more than add docker-compose files. Do you think you may modify this?

**Review by lijamie98 — CHANGES_REQUESTED** [review]

**Comment by lijamie98** [comment]
Can you also add the e2e tests to the GH actions?

File: docker-compose/common-services/Dockerfile (hunk 1 above)
Comment (paulormnas) [reply docker-compose/common-services/Dockerfile:1]: Yeah, it basically the same but without the `DATADOG` part.
I'll try to run the tests with the one that you pointed. If everything run ok, I remove the one I created.

File: CHANGELOG.md (hunk 2 above)
Comment (paulormnas) [reply CHANGELOG.md]: Updated to include information about Github workflows and automation of e2e tests

File: docker-compose/common-services/Dockerfile (hunk 1 above)
Comment (paulormnas) [reply docker-compose/common-services/Dockerfile]: I removed the Docker file I created. The one that already exists in repo works fine

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # Changelog
 ## 1.2.12
-* Add docker-compose files to run e2e tests with containers
+* Add docker-compose files to run e2e tests with containers and update Github Actions and Makefile to run automated e2e tests when merging new features to branch main.
```
Comment (lijamie98) [inline CHANGELOG.md:4]:
```suggestion
* Added Docker Compose files to facilitate the running of end-to-end tests with containers. Updated GitHub Actions and Makefile to automatically execute e2e tests upon merging new features into the main branch.
```

Comment (lijamie98) [reply CHANGELOG.md:4]: by chatGPT 🤣

**Pull request closed without merging**